library/analytics/src/main/java/org/cru/godtools/analytics/AnalyticsService.java

```diff
@@ -118,6 +134,18 @@ private void trackScreenViewInAdobe(@NonNull final String screen) {
         }
     }
 
+    @AnyThread
+    private void trackScreenViewInSnowPlow(@NonNull final String screen)
+    {
+        final Activity activity = mActiveActivity != null ? mActiveActivity.get() : null;
+        if (activity != null) {
+            mAnalyticsExecutor.execute(() -> {
+                mSnowPlowTracker.track(ScreenView.builder().name(screen).build());
+                mPreviousScreenName = screen;
```

Comment:
`mPreviousScreenName` is something specific for Adobe Analytics. do you use it for Snowplow anywhere? if not I would just leave updating it in the adobe analytics executor task, and not have it here.

Reply:
Nope, I can remove it.